erpnext/agriculture/doctype/plant_batch/plant_batch.py

```diff
@@ -52,6 +55,25 @@ def append_to_child(self, obj_to_append):
 
 		self.save()
 
+	def set_project_dates(self,periods):
+		if self.project:
+			doc = frappe.get_doc("Project", self.project)
+			doc.expected_start_date = self.start_date
+			doc.expected_end_date = frappe.utils.add_days(doc.expected_start_date, periods)
+			doc.save()
+			self.end_date = doc.expected_end_date
+
+	def set_task_dates(self,periods):
+		if self.project and (self.start_date or self.end_date):
+			tasks = frappe.get_all("Task", {"project":self.project})
+			for task in tasks:
+				doc = frappe.get_doc("Task", task.name)
+				if self.start_date:
+					doc.exp_start_date = self.start_date
+				if self.end_date:
+					doc.exp_end_date = frappe.utils.add_days(self.start_date, periods)
+				doc.save()
```

Comment:
- These lines of code are duplicated in `project.py`, just run this function when project is being saved.